test/integration/test_utils.mjs

```diff
@@ -252,26 +252,19 @@ async function waitForTextLayer(page) {
 }
 
 async function scrollIntoView(page, selector) {
+  const promise = page.evaluate(
+    () =>
+      new Promise(resolve => {
+        document
+          .getElementById("viewerContainer")
+          .addEventListener("scrollend", resolve, { once: true });
+      })
+  );
   await page.evaluate(sel => {
     const element = document.querySelector(sel);
     element.scrollIntoView({ behavior: "instant", block: "start" });
   }, selector);
-  await page.waitForFunction(
-    sel => {
-      const toolbarHeight = document
-        .querySelector("#toolbarContainer")
-        .getBoundingClientRect().height;
-      const element = document.querySelector(sel);
-      const { top, bottom } = element.getBoundingClientRect();
-      return (
-        Math.abs(top) < toolbarHeight + 100 ||
-        Math.abs(bottom - window.innerHeight) < 100 ||
-        (top > toolbarHeight && bottom < window.innerHeight)
-      );
-    },
-    {},
-    selector
-  );
+  await promise;
```

Comment:
Would it be possible to just `await page.evaluate` here so we don't need to define the logic above and get rid of the intermediate `promise` variable? Maybe even better: can't we just put this logic in the existing `await page.evaluate` block right after the `element.scrollIntoView` call? That way this function becomes smaller and easier to read. We'd end up with something like this:

```js
async function scrollIntoView(page, selector) {
  await page.evaluate(sel => {
    const element = document.querySelector(sel);
    element.scrollIntoView({ behavior: "instant", block: "start" });
    return new Promise(resolve => {
        document
          .getElementById("viewerContainer")
          .addEventListener("scrollend", resolve, { once: true });
      });
  }, selector);
}
```